.github/actions/build-image/action.yaml

```diff
@@ -59,9 +62,18 @@ runs:
       with:
         aws-region: ${{ env.aws_region }}
         role-to-assume: ${{ env.aws_gha_role }}
+    - name: Detect ECR type
+      shell: bash
+      run: |
+        echo "ECR_TYPE=private" >> $GITHUB_ENV
```

Comment:
Your TF codes doesn't have perms to push to private repos. Does it make sense to check ECR type then?